### PR TITLE
feat: honor partial Telegram reply quotes

### LINKED ISSUE #7938: [Feature] Telegram 适配器：支持「部分引用」(TextQuote)，让模型只看到用户实际选中的片段
### Description / 描述

Telegram 自 **Bot API 7.0**（[2023-12-29 changelog](https://core.telegram.org/bots/api-changelog#december-29-2023)）起支持「部分引用」：

> Added the ability to quote a part of the replied message.
> Added the class [TextQuote](https://core.telegram.org/bots/api#textquote) and the field *quote*

精确锚点：

- 协议层 `TextQuote` 类型：<https://core.telegram.org/bots/api#textquote> （字段：`text`、`position`、`entities?`、`is_manual?`）
- 协议层 `Message.quote` 字段：<https://core.telegram.org/bots/api#message>（描述："For replies that quote part of the original message, the quoted part of the message"）
- SDK `telegram.TextQuote` 类（v20.8+）：<https://docs.python-telegram-bot.org/en/stable/telegram.textquote.html>
- SDK `telegram.Message.quote` 属性（v20.8+）：<https://docs.python-telegram-bot.org/en/stable/telegram.message.html#telegram.Message.quote>

项目依赖为 `python-telegram-bot>=22.6`，字段可直接使用。

**现状**：`tg_adapter.convert_message` 只读取 `update.message.reply_to_message` 并递归转成完整的 `AstrBotMessage`，**完全忽略 `update.message.quote`**。详见 master 上的代码块（pinned to `cf6d586`）：

- <https://github.com/AstrBotDevs/AstrBot/blob/cf6d586eb916917e3119fe5084efb51c2e4bacf7/astrbot/core/platform/sources/telegram/tg_adapter.py#L496-L520>

随后 `_process_quote_message` 拼装 `<Quoted Message>...</Quoted Message>` 时拿到的也是整条原消息全文：

- <https://github.com/AstrBotDevs/AstrBot/blob/cf6d586eb916917e3119fe5084efb51c2e4bacf7/astrbot/core/astr_main_agent.py#L692-L771>

结果：模型看到的引用上下文与用户在 Telegram 客户端里实际选中的内容不一致。

### Use Case / 使用场景

群里一条很长的消息（文章片段、多条 FAQ、几行报错），用户只想就**其中一句**追问 Bot：

1. TG 客户端选中那一句 → Reply → 输入新问题发送
2. 客户端 UI 上只显示那一句作为 quote
3. **期望**：模型只看到那一句作为引用上下文
4. **实际**：模型看到整条原消息全文，「只问这一句」的意图被稀释，回答容易跑偏到原消息其它部分

Image

这里我是只想让他看public-api这个仓库， 所以特意quote这部分， 但模型看到的还是整条message

Image

### Willing to Submit PR? / 是否愿意提交 PR？

- [x] 是的，我愿意提交 PR。

### Code of Conduct

- [x] 我已阅读并同意遵守该项目的 [行为准则](https://docs.github.com/zh/site-policy/github-terms/github-community-code-of-conduct)。

### PR DESCRIPTION
**Pull request by Last-emo-boy**
Fixes #7938

### Modifications / 改动点

- Use Telegram TextQuote.text as the exact quoted reply content.
- Keep the full replied message as the fallback when no non-empty partial quote exists.
- Add regression coverage with emoji and a non-zero UTF-16 quote position.

- [x] This is NOT a breaking change. / 这不是一个破坏性变更。

### Screenshots or Test Results / 运行截图或测试结果

- uv run pytest -q tests/test_telegram_adapter.py: 12 passed
- ruff format --check .: passed
- ruff check .: passed

---

### Checklist / 检查清单

- [x] The feature was discussed with a maintainer in the linked issue.
- [x] My changes have been well-tested, and verification results are provided above.
- [x] No new dependencies are introduced.
- [x] My changes do not introduce malicious code.

## Summary by Sourcery

Honor Telegram partial reply quotes when constructing reply components while preserving existing behavior as a fallback.

New Features:
- Support using Telegram reply quote text as the replied message content when present.

Enhancements:
- Fallback to the full replied message when no non-empty quote text is available in Telegram replies.

Tests:
- Add Telegram adapter tests covering partial quote handling, empty or missing quote text, and emoji with non-zero UTF-16 quote positions.

### CODE REVIEW

File: tests/test_telegram_adapter.py [hunk 1]
```diff
@@ -82,6 +82,79 @@ def _build_context() -> MagicMock:
...
+    assert reply.message_str == quoted_text
+    assert reply.text == quoted_text
+    assert reply.chain is not None
+    assert len(reply.chain) == 1
+    assert isinstance(reply.chain[0], Comp.Plain)
+    assert reply.chain[0].text == quoted_text
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("quote_text", [None, ""])
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply tests/test_telegram_adapter.py:123]: A non-empty TextQuote is treated as the exact user-selected quote. Trimming whitespace would alter that content; the regression coverage intentionally falls back only for absent or empty quote text.

File: astrbot/core/platform/sources/telegram/tg_adapter.py [hunk 2]
```diff
@@ -506,15 +506,22 @@ def _apply_caption() -> None:
             reply_abm = await self.convert_message(reply_update, context, False)
 
             if reply_abm:
+                quote_text = update.message.quote.text if update.message.quote else None
+                reply_chain = reply_abm.message
+                reply_message_str = reply_abm.message_str
+                if quote_text:
+                    reply_chain = [Comp.Plain(quote_text)]
+                    reply_message_str = quote_text
```
[the remark this answers is not in the document]
Comment (Last-emo-boy) [reply astrbot/core/platform/sources/telegram/tg_adapter.py:514]: The project requires python-telegram-bot >= 22.6, where Message.quote is part of the supported contract, so direct access is intentional. TextQuote.text is the exact selected text and is preserved verbatim.

File: astrbot/core/platform/sources/telegram/tg_adapter.py [hunk 3]
```diff
@@ -506,15 +506,22 @@ def _apply_caption() -> None:
             reply_abm = await self.convert_message(reply_update, context, False)
 
             if reply_abm:
+                quote_text = update.message.quote.text if update.message.quote else None
```
Comment (Soulter) [inline astrbot/core/platform/sources/telegram/tg_adapter.py:509]: hi, does update.message.quote exists in the  previous version of telegram SDK AstrBot using?

I have not double check yet.

Comment (Last-emo-boy) [reply astrbot/core/platform/sources/telegram/tg_adapter.py:509]: checked.
It does appear cuz Message.quote appeared since [v20.8](https://docs.python-telegram-bot.org/en/v22.0/telegram.message.html)
whereas Astrbot requirements were `python-telegram-bot>=22.0` and bumped to `>=22.6` in PR #5726 
so the attribute exists in both the previous and current supported versions. When a message has no partial quote, its value is simply None, so direct access should be safe.

**Review by Soulter — APPROVED** [review]
